packages/application-shell/src/components/application-shell/application-shell.spec.js

```diff
@@ -1,508 +1,446 @@
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import { LOGOUT_REASONS } from '@commercetools-frontend/constants';
 import { reportErrorToSentry } from '@commercetools-frontend/sentry';
-import { AsyncLocaleData } from '@commercetools-frontend/i18n';
-import { ApplicationContextProvider } from '@commercetools-frontend/application-shell-connectors';
-import * as appShellUtils from '../../utils';
-import ConfigureIntlProvider from '../configure-intl-provider';
-import ProjectContainer from '../project-container';
-import FetchUser from '../fetch-user';
-import FetchProject from '../fetch-project';
-import AppBar from '../app-bar';
-import NavBar, { LoadingNavBar } from '../navbar';
+import { useIsAuthorized } from '@commercetools-frontend/permissions';
+import { MaintenancePageLayout } from '@commercetools-frontend/application-components';
+import LockedDiamondSVG from '@commercetools-frontend/assets/images/locked-diamond.svg';
+import {
+  createGraphqlMockServer,
+  mocksForMc,
+  UserMock,
+  ProjectMock,
+} from '../../../../../graphql-test-utils';
+import selectProjectKeyFromUrl from '../../utils/select-project-key-from-url';
+import { createApolloClient } from '../../configure-apollo';
+import { STORAGE_KEYS } from '../../constants';
 import ApplicationShellProvider from '../application-shell-provider';
-import { getBrowserLocale } from '../application-shell-provider/utils';
-import ApplicationShell, {
-  RestrictedApplication,
-  MainContainer,
-} from './application-shell';
+import ApplicationShell from './application-shell';
 
 jest.mock('@commercetools-frontend/sentry');
-jest.mock('../../utils');
+jest.mock('../../utils/select-project-key-from-url');
 
 const createTestProps = props => ({
-  applicationMessages: {
-    en: { 'CustomApp.title': 'Title en' },
-    'en-US': { 'CustomApp.title': 'Title' },
-    de: { 'CustomApp.title': 'Titel' },
-  },
   environment: {
     applicationName: 'my-app',
     frontendHost: 'localhost:3001',
     mcApiUrl: 'https://mc-api.commercetools.com',
     location: 'eu',
     env: 'development',
     cdnUrl: 'http://localhost:3001',
-    servedByProxy: false,
+    servedByProxy: 'false',
+    enableSignUp: 'true',
   },
+  featureFlags: {},
+  defaultFeatureFlags: {},
   trackingEventWhitelist: {},
-  render: jest.fn(),
-  notificationsByDomain: {
-    global: [],
-    page: [],
-    side: [],
-  },
-  showNotification: jest.fn(),
-  showApiErrorNotification: jest.fn(),
-  showUnexpectedErrorNotification: jest.fn(),
   onRegisterErrorListeners: jest.fn(),
+  onMenuItemClick: jest.fn(),
+  applicationMessages: {
+    en: { 'CustomApp.title': 'Title en' },
+    'en-US': { 'CustomApp.title': 'Title' },
+    de: { 'CustomApp.title': 'Titel' },
+  },
   ...props,
 });
 
-const testLocaleData = {
-  isLoading: false,
-  locale: 'en',
-  messages: { 'AppKit.title': 'Title en', 'CustomApp.title': 'Title en' },
+const mockSuperfluousRequests = () => {
```

Comment:
Nit: but should we extract this somehow? Feels like we might have more in the future, right?

Reply:
I guess..should I put it in the graphql-test-utils?

Reply:
Yeah could help to just have a `applyExternalNetworkRequests` there or something.